sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -419,14 +441,44 @@ class StreamExecution(
             SQLConf.SHUFFLE_PARTITIONS.key, shufflePartitionsToUse.toString)
         }
 
-        logDebug(s"Found possibly unprocessed offsets $availableOffsets " +
-          s"at batch timestamp ${offsetSeqMetadata.batchTimestampMs}")
+        offsetLog.get(latestBatchId - 1).foreach { lastOffsets =>
+          committedOffsets = lastOffsets.toStreamProgress(sources)
```

Comment:
add a comment on what you are trying to do here. also if this is part of the `// First assume that we are re-executing the latest batch in the offset log` then might as well move this up there if possible.